filclient.go

```diff
@@ -1754,10 +1820,6 @@ func (fc *FilClient) RetrieveContentFromPeerWithProgressCallback(
 		finishedTime := time.Now()
 		retrievalState.FinishedTime = finishedTime
 
-		// Otherwise publish a retrieval event success
-		retrievalEvent := rep.RetrievalEvent{Code: rep.RetrievalEventSuccess, Status: "data transfer for retrieval complete"}
-		fc.retrievalEventPublisher.Publish(retrievalEvent, retrievalState)
-
```

Comment:
pushed this block down, out of the callback, so we can attach additional success data to the state before reporting the event